### PR TITLE
fix memory leak

### PR DESCRIPTION
**Pull request by vbmithr**
cc @pdonadeo cc @rgrinberg

### CODE REVIEW

**Comment by vbmithr** [comment]
This seems to at least mitigate the memory leak issue I was experiencing using `cohttp/async`.
I'm pretty sure there was an issue when using `async_ssl` due to closing an `ssl_connect` pipe not cleaning up the underlying writer. I would be happy if anybody can review this.

File: lib/conduit_async_ssl.ml [hunk 1]
```diff
@@ -45,15 +44,27 @@ let ssl_connect ?version ?name ?ca_file ?ca_path ?session ?verify
   >>= fun conn ->
   verify_connection conn >>= function
   | false ->
+    Ssl.Connection.close conn ;
+    Pipe.close_read app_rd ;
+    Writer.close w >>= fun () ->
```
Comment (rgrinberg) [inline lib/conduit_async_ssl.ml]: How come you're closing the reader in 1 case and the pipe in the other? Seems a bit asymmetrical.

**Comment by rgrinberg** [comment]
This is a good change that I've experimented with myself, but I don't think it will affect the memory leak one way or the other. This seems to be a fix for a potential FD leak. But I don't anyone observed FD leaks with this library yet.

**Comment by vbmithr** [comment]
On 03/02/2017 15:31, Rudi Grinberg wrote:
> How come you're closing the reader in 1 case and the pipe in the other?
> Seems a bit asymmetrical.

The other is automatically closed by Ssl.Connection.close closing its
pipes. See related Async code.

**Pull request merged**